### PR TITLE
Introducing mesonbuild for Cutter

### PR DESCRIPTION
**Pull request by pelijah**
This PR based on https://github.com/radare/radare2/pull/9317.

Creating distribution on Windows:

1. Call `vcvarsall.bat %arch%`

1. Add path to qmake to PATH

1. Execute: `python meson.py --dist=%dist_dir%`

### CODE REVIEW

**Comment by xarkes** [comment]
Can you add a build for Appveyor by modifying the `.appveyor.yml` file so we can actually see it works?

**Comment by pelijah** [comment]
Yep, ninja fails at installing step on my machine too...

**Comment by pelijah** [comment]
lol I forgot about "original" meson.py for qmake build.

**Comment by xarkes** [comment]
What do you mean "original"? I'm not sue to understand why you renamed the file to `qmeson.py`.

**Comment by pelijah** [comment]
Oh just don't pay attention. It would be better if I go to sleep right now.

**Comment by xarkes** [comment]
I merged on radare2, I think you should update the submodule.

**Comment by pelijah** [comment]
Don't merge it! Qmake build fails.

**Comment by pelijah** [comment]
Well  it was bad idea to use mesonbuild directly as python module. I think it would be better to return to previous way of invoking meson via _subprocess.call()_.

**Comment by xarkes** [comment]
Dirty but if you say so.

**Comment by pelijah** [comment]
I think current build problem is related with this internal function https://github.com/mesonbuild/meson/blob/master/mesonbuild/mesonlib.py#L26.

**Comment by pelijah** [comment]
Can you merge https://github.com/radare/radare2/pull/9344?

**Comment by pelijah** [comment]
I have no idea why it works with renamed ./meson.py.

**Comment by xarkes** [comment]
Add debugging information like `dir` or `pwd` before calling meson

**Comment by pelijah** [comment]
I don't understand why bu it works with `python %CD%\sys\meson.py`. Too much undefined behaviour...

**Comment by xarkes** [comment]
Lol ok. I'll review and merge later/tomorrow.

**Comment by pelijah** [comment]
I there a way to try to build cutter on appveyour with this commit https://github.com/radare/radare2/pull/9344 without merging it to master?

**Comment by xarkes** [comment]
I don't think so. Unless you change the submodule url.

**Comment by pelijah** [comment]
I will add some more commits to this PR.

**Comment by pelijah** [comment]
You can review it.

**Comment by xarkes** [comment]
I did already :p

File: .appveyor.yml [hunk 1]
```diff
@@ -22,23 +22,43 @@ environment:
   QT64PATH: 'C:\Qt\5.10.0\msvc2015_64'
   VSVARSALLPATH: 'C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat'
   matrix:
+    # Build: qmake vs2015 x86 shared
     - BITS: 32
+      QMAKE: 1
+    # Build: qmake vs2015 x64 shared
     - BITS: 64
+      QMAKE: 1
```
Comment (xarkes) [inline .appveyor.yml]: Missing ARTIFACT_SUFFIX making the artifact name weird.

**Comment by xarkes** [comment]
After that I'll merge

**Comment by pelijah** [comment]
Fixed

**Comment by xarkes** [comment]
Great thank you.

**Pull request merged**